### PR TITLE
Issue 1960: Simple retention system test

### LINKED ISSUE #1960: Write system tests to exercise all stream workflows like update, truncate, seal delete
**Problem description**
Author system tests to exercise different controller workflows.

**Problem location**

**Suggestions for an improvement**

### PR DESCRIPTION
**Pull request by Prabha-Veerubhotla**
**Change log description**
1. New system test is added, with focus on `retention`.
2. Controller is configured with `retention frequency` of `2 * 60 s`.
**Purpose of the change**
We need system tests where controller workflows like truncate, update is exercised.
**What the code does**
Fixes #1960 
Adds a very simple retention test, where the controller is configured with `retention frequency` of  2mins, a stream is created with a retention policy of 1 min, an event is written to a stream and read after 4 mins of sleep. It is verified that data is lost. The reader only reads `null` event.
**How to verify it**
Run system tests.

### CODE REVIEW

**Comment by Prabha-Veerubhotla** [comment]
>erroneous exit, system code should be negative to indicate failure

corrected

File: test/system/src/test/java/io/pravega/test/system/RetentionTest.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        //create a reader
+        ReaderGroupManager groupManager = ReaderGroupManager.withScope(SCOPE, controllerURI);
+        groupManager.createReaderGroup(READER_GROUP, ReaderGroupConfig.builder().startingTime(0).build(),
+                Collections.singleton(STREAM));
+        EventStreamReader<String> reader = clientFactory.createReader(UUID.randomUUID().toString(),
+                READER_GROUP,
+                new JavaSerializer<>(),
+                ReaderConfig.builder().build());
+
+        //read the same event
```
Comment (shiveshr) [inline test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: the comment seems confusing.. you could say - try reading the event that was written earlier. Expectation is it should have been truncated and we should find stream to be empty.

File: test/system/src/test/java/io/pravega/test/system/RetentionTest.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        try {
+            String readEvent = reader.readNextEvent(6000).getEvent();
+            log.debug("Reading event: {} ", readEvent);
+            assertEquals(null, readEvent);
+        }  catch (ReinitializationRequiredException e) {
+            log.error("Unexpected request to reinitialize {}", e);
+            System.exit(-1);
```
Comment (shrids) [inline test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: We could use an assert.fail() here.

File: test/system/src/test/java/io/pravega/test/system/RetentionTest.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        writer.writeEvent("", writeEvent);
+        writer.flush();
+        log.debug("Writing event: {} ", writeEvent);
+
+        //sleep for 4 mins
+        Exceptions.handleInterrupted(() -> Thread.sleep(4 * 60 * 1000));
+
+        //create a reader
+        ReaderGroupManager groupManager = ReaderGroupManager.withScope(SCOPE, controllerURI);
+        groupManager.createReaderGroup(READER_GROUP, ReaderGroupConfig.builder().startingTime(0).build(),
```
Comment (shrids) [inline test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: we could disable automatic checkpoints.

File: test/system/src/test/java/io/pravega/test/system/RetentionTest.java (hunk 1 above)
Comment (Prabha-Veerubhotla) [reply test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: fixed.

File: test/system/src/test/java/io/pravega/test/system/RetentionTest.java (hunk 2 above)
Comment (Prabha-Veerubhotla) [reply test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: done. Failing the test if we get such exception.

File: test/system/src/test/java/io/pravega/test/system/RetentionTest.java (hunk 3 above)
Comment (Prabha-Veerubhotla) [reply test/system/src/test/java/io/pravega/test/system/RetentionTest.java]: added.

**Pull request merged**